guardian/managers.py

```diff
@@ -58,7 +58,11 @@ def assign_perm(self, perm, user_or_group, obj, origin=None):
             permission = perm
 
         kwargs = self._perm_kwargs(permission, user_or_group, obj, ctype, origin)
-        obj_perm, _ = self.get_or_create(**kwargs)
+        # Note that a get or create will fail with more than one result if origin is not present and multiple objects with an origin exist, however if origin is present and None, then it will also fail because None is not an origin. So we check in 2 steps.
+        obj_perm = self.filter(**kwargs).first()
+        if not obj_perm:
+            obj_perm = self.create(**kwargs)
+
```

Comment:
This is the fix for the bug, get_or_create does not work because origin cannot be None, which is a legitimate value.